src/server/services/dbManager.ts

```diff
@@ -1,6 +1,6 @@
 import { UnitType } from '../../utils/enums';
-import { Unit } from '../../utils/apiTypes';
-import { getAllUnits as dbGetAllUnits, isDBConnected } from '../../database/sqlServer/dbService';
+import * as dbService from '../../database/sqlServer/dbService';
+import Unit from '../../utils/Unit';
```

Comment:
Is /utils/Unit newly created? I can't find it in the Files Changed. What is its difference between the one in /utils/apiTypes?

Reply:
Since constructor is required for most of the cases, I have migrate some of the internal types to class to make the coding easier. I have asked Kyle to help us to migrate other internal types to class and all internal types should be deprecated inside the apiTypes.

Reply:
Just curious when will constructor be needed?

Reply:
1) When creating testcases / Mock for Unit, Question
2) When constructing a Question from a question template

Reply:
You can create an object that confirms to the desired type like:
````
const res: QuestionType = {
  question: "What is 1 + 1?",
  choices: [
    "1",
    "3",
    "2",
    "4",
  ],
  correctChoiceIndex: 2
};
````
I think it will be better to use object with type instead of class because it can automatically parse to JSON and there are no member functions and private variables in the class.

Reply:
Thanks, I will stick with your suggestion

Reply:
Great! Thanks.